bridges/SitemapBridge.php

```diff
@@ -0,0 +1,152 @@
+                'required' => true
+            ],
+            'content_cleanup' => [
+                'name' => '[Optional] Content cleanup: List of items to remove',
+                'title' => 'Selector for unnecessary elements to remove inside article contents.',
+                'exampleValue' => 'div.ads, div.comments',
+            ],
+            'title_cleanup' => [
+                'name' => '[Optional] Text to remove from article title',
+                'title' => 'Specify here some text from page title that need to be removed, e.g. " | BlogName".',
+                'exampleValue' => ' | BlogName',
+            ],
+            'site_map' => [
+                'name' => '[Optional] sitemap.xml URL',
+                'title' => <<<EOT
+                    By default, the bridge will analyze robots.txt to find out URL for sitemap.xml.
+                    Alternatively, you can specify here the direct URL for sitemap XML.
+                    The sitemap.xml file must have <loc> and <lastmod> fields for the bridge to work:
+                    Eg. <url><loc>https://article/url</loc><lastmod>2000-12-31T23:59Z</lastmod></url>
+                    <loc> is feed item URL, <lastmod> for selecting the most recent entries.
+                    EOT,
+                'exampleValue' => 'https://example.com/sitemap.xml',
+            ],
+            'limit' => self::LIMIT
+        ]
+    ];
+
+    public function collectData()
+    {
+        $url = $this->getInput('home_page');
+        $url_pattern = $this->getInput('url_pattern');
+        $content_selector = $this->getInput('content_selector');
+        $content_cleanup = $this->getInput('content_cleanup');
+        $title_cleanup = $this->getInput('title_cleanup');
+        $site_map = $this->getInput('site_map');
+        $limit = $this->getInput('limit');
+
+        $this->feedName = $this->getPageTitle($url, $title_cleanup);
+        $sitemap_url = empty($site_map) ? $url : $site_map;
+        $sitemap_xml = $this->getSitemapXml($sitemap_url, !empty($site_map));
+        $links = $this->sitemapXmlToList($sitemap_xml, $url_pattern, empty($limit) ? 10 : $limit);
+
+        if (empty($links) && empty(sitemapXmlToList($sitemap_xml))) {
+            returnClientError('Could not retrieve URLs with Timestamps from Sitemap: ' . $sitemap_url);
+        }
+
+        foreach ($links as $link) {
+            $this->items[] = $this->expandEntryWithSelector($link, $content_selector, $content_cleanup, $title_cleanup);
+        }
+    }
+
+    /**
+     * Retrieve site map from specified URL
+     * @param string $url URL pointing to any page of the site, e.g. "https://example.com/blog" OR directly to the site map e.g. "https://example.com/sitemap.xml"
+     * @param string $is_site_map TRUE if the specified URL points directly to the sitemap XML
+     * @return object Sitemap DOM (from parsed XML)
+     */
+    protected function getSitemapXml(&$url, $is_site_map = false)
```

Comment:
dont pass reference

Reply:
The reference here is for updating the `$sitemap_url` in the caller context.
Before the call, `$sitemap_url` is e.g. `https://example.com/`
After the call, `$sitemap_url` is e.g. `https://example.com/seo/sitemap.xml`
I did this to show a more meaningful error message here (line 75):
```php
returnClientError('Could not retrieve URLs with Timestamps from Sitemap: ' . $sitemap_url);
```
I could remove the reference, and instead would either not update the variable (less meaningful error message), or return it as part of return value (less practical/less readable for other maintainers). What do you think?